lib/ex_aws/sns.ex

```diff
@@ -148,10 +148,18 @@ defmodule ExAws.SNS do
     params |> Map.put(param_root <> ".Value.BinaryValue", value)
   end
 
+  defp put_value(params, param_root, {:string_array, value}) do
+    params |> Map.put(param_root <> ".Value.StringValue", jsonify_array(value))
+  end
+
   defp put_value(params, param_root, {_, value}) do
     params |> Map.put(param_root <> ".Value.StringValue", value)
   end
 
+  defp jsonify_array(array) do
+    "[" <> Enum.map_join(array, ",", &("\"#{&1}\"")) <> "]"
```

Comment:
I initially wanted to use Poison, but it's only included in `dev` / `test` dependencies, so I preferred this trivial serialisation code, which is not rocket science. Open to change.

Reply:
For this you can use `config[:json_codec].encode(array)` to access whichever json parser the user has configured.

Reply:
Thanks for the feedback! I wasn't aware that it was a global ExAws configuration parameter. It's better indeed. I wasn't able to access the `config` object though. I don't think it is passed to ExAws.SNS, and adding it to the `publish()` parameters would likely be breaking changes. Instead, I'm reading it directly from the Elixir environment. Is this ok? Do you think there's a failure case where `:json_codec` is not defined?

Tell me if I missed something or if you have other design preferences. I'll update accordingly.

Reply:
Nah, that should be fine, thanks. Almost anyone using ExAws should have `json_codec` set - it's relied on by heaps of other parts of the system.